### ISSUE TITLE
Markdown component 

### ISSUE THREAD

**Issue by ycmjason** [vuejs/vuepress#222]
Hello, 

I think it would be awesome if we could have a markdown vue component that could be used in the markdown pages. 

This way we could module our very long markdown into smaller sections.

Thoughts?

E.g.

```
# My very long md

This is a freaking long md...

<vp-md src="./sectionA.md"/>

<vp-md src="./sectionB.md"/>

<vp-md src="./sectionC.md"/>
```

**Comment by yyx990803** [comment]
I don't think this has to be a component, can simply be a comment-based directive:

``` md
<!-- include ./sectionA.md -->
```

And then parsed and handled directly inside `lib/webpack/markdownLoader.js`.

**Comment by ycmjason** [comment]
I will look into that tonight if I got the time.

**Comment by ycmjason** [comment]
An attempt in here https://github.com/vuejs/vuepress/pull/227.

**Comment by MartinMuzatko** [comment]
There is already https://www.npmjs.com/package/markdown-it-include
But I see a few problems with this:

1. no vue templating within those included markdown files.
2. partials have to be excluded from getting rendered by vuepress

**Comment by ycmjason** [comment]
@MartinMuzatko my attempt at #227 should be able to load vue templating.

Yet, the partials are also being rendered. Maybe we could make it so that vuepress will ignore `.partial.md` files and not render them.

**Comment by eyleron** [comment]
How about the convention in other template languages or SCSS where filenames with preceding underscore don't get rendered?
``` _layout-header.md
```

**Comment by ycmjason** [comment]
@eyleron I am not so sure about this as @yyx990803 and @ulivz haven't got time to advice on this.

**Comment by davemacdo** [comment]
I know this is an old Issue, but it's something I would definitely use. I did want to point out that there are already several syntaxes for this in various markdown flavors. Personally, I'm used to using Mulitmarkdown's [file transclusion syntax](http://fletcher.github.io/MultiMarkdown-5/transclusion.html) in a lot of my markdown already: 

```
# Parent Markdown Stuff

{{path/to/included.md}}
```

See also: 

- Brett Terpstra's [multi-file document syntax](http://marked2app.com/help/Multi-File_Documents.html), which has been adopted by other markdown processors
- iA Writer's [Markdown Content Blocks](https://github.com/iainc/Markdown-Content-Blocks)

A lot of the content I'm including in VuePress sites exists as Markdown already, so the more it can mimic the behavior of other renderers, the better.

**Comment by ycmjason** [comment]
@davemacdo 
Please see this: https://github.com/vuejs/vuepress/pull/227

There is a plan for rolling this out.

https://github.com/vuejs/vuepress/issues/421

In 1.0.0

**Comment by davemacdo** [comment]
Thanks. I missed that in my search. I'll take my comment there.

**Comment by rajaraodv** [comment]
This would be EXCELLENT! We have several "technical docs" people who want to reuse markdowns as "templates" and not learn Vue. I'm really hoping that this features is added! BTW, thanks a lot of creating Vuepress!

**Comment by ycmjason** [comment]
@rajaraodv 
No worries. Now vuepress next is in alpha, i will just try to make this a plugin tonight. 👍

**Comment by ycmjason** [comment]
@rajaraodv 
I have got an implementation already. But I will wait until Next version is available then I will make this a plugin.

**Comment by rajaraodv** [comment]
@ycmjason Awesome!! Is there any way to use it in the current version? Context: I'm showing off how cool vuepress is to folks and one question I'm getting is that the images are too small. I just need to hack it for the time being.

**Comment by bretterer** [comment]
@ulivz see https://github.com/vuejs/vuepress/issues/1173   it seems to be broken.

**Comment by coffeephile** [comment]
@ulivz I wanted to include a markdown file from another markdown file using `<Content />` but I had trouble getting it to work.

I came up with a working solution by kind of exposing `findPageForPath()` to the markdown scope with the following mixin.
```
// enhanceApp.js

import { findPageForPath } from '@app/util'

export default ({
  Vue
}) => {
  Vue.mixin({
    methods: {
      getPageKey(pages, path) {
        const result = findPageForPath(pages, path)
        return result.key
      }
    }
  })
}
```

```
<!-- my-markdown-file.md -->

<Content :page-key="getPageKey($site.pages, '/path/to/my-other-markdown-file/')" />
```

This solution works, however I am wondering if there is a better way to achieve this.

Any recommendations appreciated. 🙇

**Comment by fsodano** [comment]
Is there a better way to achieve this nowadays? For now I'm forced to do:

`<Content :page-key="$site.pages.find(p => p.path === '/my-page.html').key"/>`

Cheers.

**Comment by deepansh96** [comment]
I got it working with `markdown-it-include` [package](https://www.npmjs.com/package/markdown-it-include).

This is how I extended markdown in my `.vuepress/config.js`

```js
module.exports = {
    .
    .
    .
    extendsMarkdown: (md) => {
        // use more markdown-it plugins!
        md.use(require('markdown-it-include'))
    }
}
```

The key here is that I was using `v2` of vuepress and there the API has changed from `extendMarkdown` to `extendsMarkdown`

After doing this, go to the parent `.md` file where you want to inlcude child `.md` files and include them like this
```md
!!!include(./docs/your-directory/getting-started.md)!!!
```

**Comment by alexfornuto** [comment]
For those also brought here by Google looking for a solution to includes, @deepansh96's solution works, but don't forget to change `extendsMarkdown` to `extendMarkdown`.

**Comment by deepansh96** [comment]
> I got it working with `markdown-it-include` [package](https://www.npmjs.com/package/markdown-it-include).
> 
> This is how I extended markdown in my `.vuepress/config.js`
> 
> ```js
> module.exports = {
>     .
>     .
>     .
>     extendsMarkdown: (md) => {
>         // use more markdown-it plugins!
>         md.use(require('markdown-it-include'))
>     }
> }
> ```
> 
> The key here is that I was using `v2` of vuepress and there the API has changed from `extendMarkdown` to `extendsMarkdown`
> 
> After doing this, go to the parent `.md` file where you want to inlcude child `.md` files and include them like this
> 
> ```lisp
> !!!include(./docs/your-directory/getting-started.md)!!!
> ```

This is breaking on the current version of vuepress@next.
This was working in the version `^2.0.0-beta.19` but breaking in `^2.0.0-beta.59`.

The error I'm getting this this:
`Error: Dynamic require of "markdown-it-include" is not supported`

**Comment by duhdugg** [comment]
> 
> This is breaking on the current version of vuepress@next. This was working in the version `^2.0.0-beta.19` but breaking in `^2.0.0-beta.59`.
> 
> The error I'm getting this this: `Error: Dynamic require of "markdown-it-include" is not supported`

this is working for me in a "type": "module" package:

```typescript
import markdownItInclude from "markdown-it-include";
// ...
export default defineUserConfig({
  // ...
  extendsMarkdown: (md) => {
    md.use(markdownItInclude);
  }
});
```
